tests-dev/create_yml.py

```diff
@@ -0,0 +1,185 @@
+        else:
+            with open(r'ufs_test_temp.yaml', 'w') as yaml_file:
+                outputs = yaml.dump(new_yaml, yaml_file)
+                yaml_file.close()
+        file_yaml.close()
+
+def update_testyaml_n():
+    """Update test yaml file for a single test specified in -n <test_name> <compiler>
+    """
+    try:
+        SRT_NAME     = str(os.getenv('SRT_NAME'))
+        SRT_COMPILER = str(os.getenv('SRT_COMPILER'))
+    except NameError:
+        print("*** SRT_NAME or SRT_COMPILER are not given with runtime option -n! ***")
+    input_list=[SRT_NAME+" "+SRT_COMPILER]
+    update_testyaml(input_list)
+
+def update_testyaml_b():
+    """Update test yaml file for tests specified in -b <file>
+    """
+    NEW_BASELINES_FILE = str(os.getenv('NEW_BASELINES_FILE'))
+    input_list=[]
+    with open(NEW_BASELINES_FILE) as input_file:
+        for line in input_file:
+            line=line.strip('\n')
+            line=line.strip()
+            input_list.append(str(line))
+        input_file.close()
+    update_testyaml(input_list)
+
+def string_clean(str_in):
+    """Strip out RUN or COMPILE whitespace and separate with commas.
+
+    Args:
+        str_in (str): RUN or COMPILE line read in from rt.conf
+
+    Returns:
+        str: whitespace stripped and comma separated values
+    """
+    return "'"+("','".join(str_in.split()))+"'"
+
+def parse_line(str_in):
+    """Parse rt.conf line into list
+
+    Args:
+        str_in (str): RUN or COMPILE line from rt.conf
+
+    Returns:
+        list: list of RUN or COMPILE test attributes
+    """
+    build_attr = " ".join(str_in.split()).split('|')
+    build_attr = [attr.strip() for attr in build_attr]
+    return build_attr
+
+if __name__ == "__main__":
+    with open('ufs_test.yaml', 'w') as yaml_file, open("rt.conf") as conf_file:
+        for line in conf_file:
+            line = line.strip()
+            if not line:  # skip: line is blank
+                continue
+            if line.startswith("#"):  # skip: comment line
+                continue
+            if line.startswith("COMPILE"):  # COMPILE line
+                build = parse_line(line)
+                apps = build[1]
+                compiler = f"'{build[2]}'"
+                options = f"'{build[3]}'"
+                machine = build[4]
+                off_machine = None
+                on_machine = None
+                if (machine.find('-') != -1):
+                    off_machine = machine.replace("-", "").strip()
+                    off_machine = string_clean(off_machine)
+                if (machine.find('+') != -1):
+                    on_machine = machine.replace("+", "").strip()
+                    on_machine = string_clean(on_machine)                
+                yaml_file.write(f"{apps}_{build[2].strip()}:\n")
+                yaml_file.write(f"  build: \n")
+                yaml_file.write(f"    compiler: {compiler}\n")
+                yaml_file.write(f"    option: {options}\n")
```

Comment:
Is this how yaml files are supposed to be created in python by carefully writing formatted strings to a text file? Is'n the 'correct' why of writing yaml file to create data structure in memory and then 'dump' into a file?

Reply:
Intention is to provide a starting point of initial yaml file. Further update option will be needed with @ulmononian feedback.

Reply:
This is absolutely not how it should be done. This is a bad practice. I highly recommend against it.

Reply:
please, come up with an idea to convert rt.conf to yaml.